src/tools/compiletest/src/header.rs

```diff
@@ -1660,8 +1667,16 @@ fn ignore_llvm(config: &Config, path: &Path, line: &str) -> IgnoreDecision {
     IgnoreDecision::Continue
 }
 
-enum IgnoreDecision {
-    Ignore { reason: String },
-    Continue,
-    Error { message: String },
+fn handle_known_bug(config: &Config, ln: &str) -> IgnoreDecision {
+    // Crashes tests need to ICE or crash, even with `//@ known-bug`.
+    if config.mode == Mode::Crashes {
+        return IgnoreDecision::Continue;
+    }
+
+    // NOTE: `//@ known-bug` is additionally validated in test prop parsing.
+    if let Some(bugs) = config.parse_name_value_directive(ln, "known-bug") {
+        IgnoreDecision::Ignore { reason: format!("known-bug: {bugs}") }
```

Comment:
Wait...known-bug tests shouldn't be ignored? The whole point of these is to ensure that we can detect when tests change. This should not only be the case for crashes.

Reply:
> Wait...known-bug tests shouldn't be ignored? The whole point of these is to ensure that we can detect when tests change. This should not only be the case for crashes.

Hm you're right, let me think about this a bit more.

Reply:
Not sure why I thought it was a good idea to ignore `//@ known-bug` tests, that can't be right.